packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -0,0 +1,294 @@
+  const components = Array.from(structuredBody.getElementsByTagName("component"));
+
+  const problemListComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Problem list";
+  });
+
+  const treatmentPlanComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "Treatment Plan";
+  });
+
+  const medicationComponent = components.find(component => {
+    const title = component.getElementsByTagName("title")[0];
+    return title?.childNodes[0].nodeValue === "History of Medication use";
+  });
+
+  if (problemListComponent) {
+    const problemListTable = problemListComponent?.getElementsByTagName("table")[0];
+
+    addIcd10Codes(problemListComponent, problemListTable, document);
+  }
+
+  if (treatmentPlanComponent) {
+    const firstComponent = components[0];
+    structuredBody.insertBefore(treatmentPlanComponent, firstComponent);
+  }
+
+  if (problemListComponent) {
+    const reshapedComponents = Array.from(structuredBody.getElementsByTagName("component"));
+    const secondComponent = reshapedComponents[1];
+    structuredBody.insertBefore(problemListComponent, secondComponent);
+  }
+
+  if (medicationComponent) {
+    const reshapedComponents = Array.from(structuredBody.getElementsByTagName("component"));
+    const thirdComponent = reshapedComponents[2];
+    structuredBody.insertBefore(medicationComponent, thirdComponent);
+  }
+
+  return document.toString();
+};
+
+const addIcd10Codes = (
+  problemListComponent: Element,
+  problemListTable: Element,
+  document: Document
+) => {
+  addCodesToColumn(problemListTable, document);
+
+  const tableRows = problemListTable.getElementsByTagName("tbody")[0].getElementsByTagName("tr");
+
+  addCodesToRows(problemListComponent, tableRows, document);
+};
+
+const addCodesToColumn = (problemListTable: Element, document: Document) => {
+  const codeColumn = document.createElement("th");
+  codeColumn.appendChild(document.createTextNode("Codes"));
+  const columns = Array.from(problemListTable.getElementsByTagName("th"));
+
+  columns.forEach(column => {
+    if (column.childNodes[0].nodeValue === "Problem") {
+      const parentNode = column.parentNode;
+
+      parentNode?.insertBefore(codeColumn, column.nextSibling);
+    }
+  });
+};
+
+const addCodesToRows = (
+  problemListComponent: Element,
+  tableRows: HTMLCollectionOf<Element>,
+  document: Document
+) => {
+  const rows = Array.from(tableRows);
+
+  for (const row of rows) {
+    const tableCells = Array.from(row.getElementsByTagName("td"));
+    const problemNameCell = tableCells.find(cell => {
+      return cell.getAttribute("ID")?.includes("problemName");
+    });
+
+    if (problemNameCell) {
+      const entryIndex = problemNameCell.getAttribute("ID")?.split("_")[1];
```

Comment:
should the rest of the code be executed if this is undefined?

Reply:
good point